lux-lib/src/luarc/mod.rs

```diff
@@ -0,0 +1,179 @@
+    other: BTreeMap<String, serde_json::Value>,
+
+    #[serde(default)]
+    workspace: Workspace,
+}
+
+#[derive(Serialize, Deserialize, Default, PartialEq, Debug)]
+struct Workspace {
+    #[serde(default)]
+    library: Vec<String>,
+}
+
+pub fn update_luarc(config: &Config) {
+    if !config.generate_luarc() {
+        return; // do nothing
+    }
+    let project = Project::current_or_err().expect("failed to get current project");
+    let lockfile = project.lockfile().expect("should have a lockfile");
+    let luarc_path = project.luarc_path();
+    let dependency_tree = project.tree(config).expect("failed to get project tree");
+    let dependency_tree_root_relative_path = dependency_tree
+        .root()
+        .strip_prefix(project.root())
+        .expect("tree root should be a subpath of project root")
+        .to_path_buf();
+
+    let test_dependency_tree = project
+        .test_tree(config)
+        .expect("failed to get project test tree");
+    let test_dependency_tree_root_relative_path = test_dependency_tree
+        .root()
+        .strip_prefix(project.root())
+        .expect("test tree root should be a subpath of project root")
+        .to_path_buf();
+
+    // read the existing .luarc file or create a new one if it doesn't exist
+    let luarc_content = fs::read_to_string(&luarc_path).unwrap_or_else(|_| String::from("{}"));
+
+    let dependency_dirs = find_dependency_dirs(
+        &lockfile,
+        dependency_tree_root_relative_path,
+        &LocalPackageLockType::Regular,
+    );
+
+    let test_dependency_dirs = find_dependency_dirs(
+        &lockfile,
+        test_dependency_tree_root_relative_path,
+        &LocalPackageLockType::Test,
+    );
+
+    let all_dependecy_dirs: Vec<PathBuf> = dependency_dirs
+        .into_iter()
+        .chain(test_dependency_dirs)
+        // make sure the paths actually exist
+        .filter(|path| fs::exists(path).is_ok_and(|exists| exists))
+        .collect();
+
+    let file = generate_luarc(luarc_content.as_str(), all_dependecy_dirs);
+
+    fs::write(&luarc_path, file)
+        .unwrap_or_else(|_| panic!("failed to write {} file", luarc_path.display()));
+}
+
+fn find_dependency_dirs(
+    lockfile: &ProjectLockfile<ReadOnly>,
+    lux_tree_base_dir: PathBuf,
+    local_package_lock_type: &LocalPackageLockType,
+) -> Vec<PathBuf> {
+    let rocks = lockfile.local_pkg_lock(local_package_lock_type).rocks();
+
+    rocks
+        .iter()
+        .map(|t| lux_tree_base_dir.join(format!("{}-{}@{}/src", t.0, t.1.name(), t.1.version())))
+        .collect()
+}
+
+fn generate_luarc(prev_contents: &str, extra_paths: Vec<PathBuf>) -> String {
+    let mut luarc: LuaRC = serde_json::from_str(prev_contents).unwrap();
+
+    // remove any preexisting lux library paths
+    luarc
+        .workspace
+        .library
+        .retain(|path| !path.starts_with(".lux/"));
+
+    for p in extra_paths {
+        let path = p.clone().into_os_string().into_string();
```

Comment:
issue: Use `to_slash_lossy` (requires a `use path_slash::PathBufExt;`).

Lua has issues with Windows path separators. `to_slash_lossy` ensures that we use unix style path separators.